fastapi_websocket_pubsub/event_broadcaster.py

```diff
@@ -29,14 +29,81 @@ class BroadcasterAlreadyStarted(EventBroadcasterException):
     pass
 
 
+class EventBroadcasterContextManger:
+    """
+    Manages the context for the EventBroadcaster
+    Friend-like class of EventBroadcaster (accessing "protected" members )
+    """
+
+    def __init__(self, event_broadcaster: "EventBroadcaster", listen: bool = True, share: bool = True) -> None:
+        """
+        Provide a context manager for an EventBroadcaster, managing if it listens to events coming from the broadcaster 
+        and if it subscribes to the internal notifier to share its events with the broadcaster
+
+        Args:
+            event_broadcaster (EventBroadcaster): the broadcaster we manage the context for.
+            share (bool, optional): Should we share events with the broadcaster. Defaults to True.
+            listen (bool, optional): Should we listen for incoming events from the broadcaster. Defaults to True.
+        """
+        self._event_broadcaster = event_broadcaster
+        self._share: bool = share
+        self._listen: bool = listen
+        self._lock = asyncio.Lock()
+
+    async def __aenter__(self):
+        async with self._lock:
+            if self._listen:
+                self._event_broadcaster._listen_count += 1
+                if self._event_broadcaster._listen_count == 1:
+                    # We have our first listener start the read-task for it (And all those who'd follow)
+                    logger.info("Listening for incoming events from broadcast channel (first listener started)")
+                    # Start task listening on incoming broadcasts
+                    self._event_broadcaster.start_reader_task()
+
+            if self._share:
+                self._event_broadcaster._share_count += 1
+                if self._event_broadcaster._share_count == 1:
+                    # We have our first publisher
+                    # Init the broadcast used for sharing (reading has its own)
+                    self._event_broadcaster._acquire_publishing_broadcast()                
```

Comment:
unless i am mistaken, this can sit at `EventBroadcaster.__init__`
there is no acquiring here, only initializing the broadcaster object.

Reply:
actually i am not even sure self._broadcast should be a class member, see my other comment

Reply:
I thought It's better no to init the DB object which might actually do some heavy work (e.g. network actions) unless there's a need for it.

Reply:
@orweis i think all the work of Broadcast is done at __aenter__ (runs connect()) and __aexit__ (runs disconnect())

Reply:
Since Broadcaster is pluggable/extendable - I think we should be defensive of how new channels might work.
For example an AWS-SQS channel might try to obtain cloud credentials, or the AWSCLI - in general you can't be sure what those plugins would do...